src/main/c/Jep/jep_exceptions.c

```diff
@@ -82,6 +72,34 @@ int process_py_exception(JNIEnv *env)
     PyErr_Fetch(&ptype, &pvalue, &ptrace);
 
     if (ptype) {
+        // let's not turn this into a Java exception if the user exited
+        if (ptype == PyExc_SystemExit) {
+            // If the exception is not normalized the exit code is in pvalue.
+            PyObject *code = pvalue;
+	    if (PyErr_GivenExceptionMatches(pvalue, ptype)) {
+                // If the exception is normalized the exit code is in the code attribute.
+                code = PyObject_GetAttrString(pvalue, "code");
+            }
+            if (code == NULL) {
+                /* 
+                 * It should be impossible to get here unless the code attribute
+                 * is missing. There is no reasonable way to handle that case so
+                 * clear the error and let the SystemExit become a JepException
+                 */
+                PyErr_Clear();
+            } else if (code == Py_None) {
+                exit(0);
+            } else if (PyLong_Check(code)) {
+                exit(PyLong_AsLong(code));
```

Comment:
I don't understand why you are calling exit.  That's not what we did before.

Reply:
That seems like the best way to maintain the current behavior. Previously we relied on PyErr_PrintEx which would eventually exit for us, since that will no longer reliably exits(see #359) it seems simplest to just exit ourselves.